client/src/components/CreatePost/CreatePost.tsx

```diff
@@ -0,0 +1,217 @@
+});
+
+// Type for the marker IDs
+type MarkerId = 'cone' | 'pothole' | 'roadDamage' | 'carAccident' | 'warningSign' | 'etc'
+
+export default function CreatePost() {
+    // State Variables
+    const [open, setOpen] = useState(false);
+    const [postName, setPostName] = useState('');
+    const [marker, setMarker] = useState<MarkerId | null>(null);
+    const [date, setDate] = useState('');
+    const [description, setDescription] = useState('');
+    const [location, setLocation] = useState('');
+
+    // Function to handle opening CreatePost component
+    const handleClickOpen = () => {
+        setOpen(true);
+        setPostName('');
+        setMarker(null);
+        setDate(new Date().toISOString().split('T')[0]);
+
+        // Check for Geolocation support
+        if (navigator.geolocation) {
+            // Ask for user permission
+            navigator.geolocation.getCurrentPosition(
+                // Success callback
+                (position) => {
+                    // Extract latitude and longitude
+                    const { latitude, longitude } = position.coords;
+                    // Set location to latitude and longitude
+                    setLocation(`Latitude: ${latitude}, Longitude: ${longitude}`);
+                },
+                // Error callback
+                (error) => {
+                    console.error('Error getting current location:', error);
+                    // Set a default location if there's an error
+                    setLocation('Default location');
+                }
+            );
+        } else {
+            console.error('Geolocation is not supported by this browser.');
+            // Set a default location if Geolocation is not supported
+            setLocation('Default location');
+        }
+        setDescription('');
+
+    }
+    // Function to handle closing CreatePost component
+    const handleClose = () => setOpen(false);
+
+    // Functions to handle input changes to each data point 
+    // (postName, marker, date, description, and location)
+    const handlePostNameChange = (value: string) => setPostName(value);
+    const handleDateChange = (value: string) => setDate(value);
+    const handleDescriptionChange = (value: string) => setDescription(value);
+    const handleLocationChange = (value: string) => setLocation(value);
+    const markerClick = (id: string) => setMarker(id as MarkerId);
+
+    // Function to handle post submission
+    const handlePost = () => {
+        // For now it console logs the data points
+        // May need to refactor for backend
+        if (postName && description && date && location && marker) {
+            console.log('Post Name:', postName);
+            console.log('Description:', description);
+            console.log('Date:', date);
+            console.log('Selected Marker:', marker);
+            console.log('Location:', location);
+            handleClose();
+        }
+    }
+
+    return (
+        <>
```

Comment:
I think you could replace this fragment with this component from MUI: https://mui.com/material-ui/react-drawer/#swipeable